synapse/push/__init__.py

```diff
@@ -130,6 +131,7 @@ def as_dict(self) -> Dict[str, Any]:
             "profile_tag": self.profile_tag,
             "pushkey": self.pushkey,
             "enabled": self.enabled,
+            "device_id": self.device_id,
```

Comment:
The method name is unfortunate here because it makes me think of https://www.attrs.org/en/stable/api.html#attrs.asdict. Not something to deal with here though.